File: src/main/scala/com/abaddon83/cardsGames/burracoGames/domainModels/burracoGames/completed/BurracoPoint.scala
package com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.completed

import com.abaddon83.cardsGames.burracoGames.domainModels.BurracoId
import com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.initialised.playerInGames.Burraco
import com.abaddon83.cardsGames.shares.decks.Card

case class BurracoPoint protected (
                         override protected val burracoId: BurracoId,
                         override protected val cards: List[Card]
                       ) extends Burraco {
}

object BurracoPoint{
  def apply(burraco: Burraco): BurracoPoint = {
    new BurracoPoint(burraco.getBurracoId(), burraco.showCards)
  }
}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/models/burracoGameendeds/BurracoPoint.kt
package com.abaddon83.burracoGame.commandModel.models.burracoGameendeds

import com.abaddon83.burracoGame.commandModel.models.burracos.Burraco
import com.abaddon83.burracoGame.commandModel.models.burracos.BurracoIdentity
import com.abaddon83.burracoGame.commandModel.models.decks.Card

data class BurracoPoint private constructor(
        override val identity: BurracoIdentity,
        override val cards: List<Card>
): Burraco() {
    constructor(burraco: Burraco): this(burraco.identity(),burraco.showCards())

}